### DISCUSSION TITLE
@this.on() seems to have been silently deprecated

### DISCUSSION THREAD

**Discussion by markjaquith** [livewire/livewire#5892] [7. Bugs (archived)]
Jetstream is doing this in the `action-message.blade.php` template which worked in Livewire v2:
 
```
x-init="@this.on('{{ $on }}', () => { clearTimeout(timeout); shown = true; timeout = setTimeout(() => { shown = false }, 2000); })"
```

(`$on` is a passed prop, which could be the name of a Livewire event it wants to listen to, to show itself for two seconds)

In Livewire 3, there does not appear to be a `.on()` on the JavaScript object, so this code tries to call an `on()` method on the Livewire PHP component, which fails.

I migrated the code to this, which works:

```
x-on:{{ $on }}.window="clearTimeout(timeout); shown = true; timeout = setTimeout(() => { shown = false }, 2000);"
```

I think this is superior syntax and it shows the power of Alpine being more tightly integrated with Livewire, but it seems worth noting this change in the docs, if it was intentional.

**Comment by joshhanley** [comment]
@markjaquith it was likely just missed, PRs to the docs to mention the `@this.on()` removal would be welcome!

Jetstream is yet to be updated for V3, but your method looks good to me!

**Reply by markjaquith** [reply]
Cool. I'll make a docs PR.

Would you also like a manual step mentioned in the upgrader?

P.S. I have a Jetstream branch that seems to be working with Livewire 3

https://github.com/markjaquith/jetstream/tree/livewire-3

**Reply by markjaquith** [reply]
Docs PR: https://github.com/livewire/livewire/pull/5939

**Reply by joshhanley** [reply]
@markjaquith thanks!

Not sure what you mean by manual step, can you explain that bit?

Also nice re jetstream! 😁

**Reply by markjaquith** [reply]
> Not sure what you mean by manual step, can you explain that bit?

The upgrader does some things for you, and for other things, it just tells you "hey, this thing changed, so here's what you have to do". It calls those manual steps.

**Comment by julesgraus** [comment]
I had the exact same issue. I think there's something more needed here. Let me give a little bit more background information first: 

In laravel jetstream there is indeed this ```UpdateProfileInformationForm``` component. It dispatched a ```saved``` event using ```$this->emit('saved');```. I've upgraded that call to ```$this->dispatch('saved');``` to make it compatible with Livewire 3.  

Now, i also did upgrade action-message.blade.php like [markjaquith](https://github.com/markjaquith) did. But....shouldnt it work without tacking on ```.window``` to the ```x-on``` listener? 

The reason why i am asking is because that does not work. And tacking on ```.window``` shows other components action messages too which listen to that event.
